src/wifi/model/dmg-wifi-mac.cc

```diff
@@ -401,7 +401,7 @@ DmgWifiMac::ScheduleServicePeriod (uint8_t blocks, Time spStart, Time spLength,
 void
 DmgWifiMac::StartServicePeriod (AllocationID allocationID, Time length, uint8_t peerAid, Mac48Address peerAddress, bool isSource)
 {
-  NS_LOG_FUNCTION (this << length << static_cast<uint16_t> (peerAid) << peerAddress << isSource << Simulator::Now ());
+  NS_LOG_FUNCTION (this << length << +peerAid << peerAddress << isSource << Simulator::Now ());
```

Comment:
why is there the "+" in front of peerAid? Is that a typo or does it mean something?

Reply:
peerAid is a uint8_t. By default, operator<< interprets it as a character printing unexpected and meaningless logs. You need to cast it or do something with it in order to properly display it in the logs, and I think adding a plus in front of it does just that